cli/tests/pcluster/test_commands.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2013-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module provides unit tests for the functions in the pcluster.commands module."""
+
+import pytest
+from botocore.exceptions import ClientError
+
+import pcluster.utils as utils
+from assertpy import assert_that
+from pcluster.commands import create_bucket_with_batch_resources
+
+
+@pytest.mark.parametrize(
+    "region,error,action",
+    [
+        ("eu-west-1", "create_bucket", "BucketAlreadyExists"),  # bucket creation failure
+        ("eu-west-1", "upload_fileobj", "ExpiredToken"),  # resources upload failure
+        ("eu-west-1", "delete", "AccessDenied"),  # bucket deletion failure
+        ("eu-west-1", None, None),
+    ],
+)
+def test_create_bucket_with_batch_resources(region, action, error, mocker, caplog):
```

Comment:
Optional: in this casa separate test functions rather than a parametrize one might have simplified the test function

Reply:
you right. splitted.